src/test/java/org/cryptomator/cryptofs/DeleteNonEmptyCiphertextDirectoryIntegrationTest.java

```diff
@@ -159,21 +161,35 @@ private Path firstEmptyCiphertextDirectory() throws IOException {
 		try (Stream<Path> allFilesInVaultDir = Files.walk(pathToVault)) {
 			return allFilesInVaultDir //
 					.filter(Files::isDirectory) //
-					.filter(this::isEmptyDirectory) //
+					.filter(this::isEmptyCryptoFsDirectory) //
 					.filter(this::isEncryptedDirectory) //
 					.findFirst() //
 					.get();
 		}
 	}
 
-	private boolean isEmptyDirectory(Path path) {
+	private boolean isEmptyCryptoFsDirectory(Path path) {
+		Predicate<Path> isIgnoredFile = p -> Constants.DIR_ID_FILE.equals(p.getFileName().toString());
 		try (Stream<Path> files = Files.list(path)) {
-			return files.filter(p -> p.getFileName().toString().equals(Constants.DIR_ID_FILE)).count() == 0;
+			return files.noneMatch(isIgnoredFile.negate());
```

Comment:
efficiency: `noneMatch` reads the whole stream, however it should be sufficient to find `anyMatch` that isn't ignored.

Reply:
Changed in [2c0b6d6](https://github.com/cryptomator/cryptofs/pull/118/commits/2c0b6d6d1db43c696b4c6bd33f24468a575c41e0)

Reply:
I was wrong, noneMatch will of course stop on first encounter of a positive match. Therefore it behaves identical to `!anyMatch(...)`.

Please undo the change in 2c0b6d6 for two reasons:
1. readability 😉
2. more importantly, `firstMatch()` relies on a sorted stream. if the stream is not already ordered (even though we should be fine in this case), this will _really_ consume the whole stream in order to sort it. What you want to use in this case is `findAny()`. In general: Always use `findAny()` except in cases where you need the `first` item of an ordered stream.

Reply:
Done [310c038](https://github.com/cryptomator/cryptofs/pull/118/commits/310c038363fe2b244bb8f991a46dc1c7d2b85a7f)